presto-main/src/main/java/com/facebook/presto/operator/aggregation/reservoirsample/ReservoirSampleStateFactory.java

```diff
@@ -0,0 +1,44 @@
+package com.facebook.presto.operator.aggregation.reservoirsample;
+
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.operator.aggregation.reservoirsample.ReservoirSampleState;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.function.AccumulatorStateFactory;
+
+import static com.facebook.presto.spi.StandardErrorCode.FUNCTION_IMPLEMENTATION_ERROR;
+import static java.lang.String.format;
+
+public class ReservoirSampleStateFactory
```

Comment:
Since the two classes here are virtually the same, you can probably just delete both the Single and Grouped variants of the class, and then for all methods here just return `ReservoirSampleState(.class)` directly